array-reduce/main.js

```diff
@@ -0,0 +1,43 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((accumulator, currentValue) => accumulator + currentValue);
+console.log(sum);
+
+const product = numbers.reduce((accumulator, currentValue) => accumulator * currentValue);
+console.log(product);
+
+const balance = account.reduce((accumulator, currentValue) => {
+  if(currentValue.type === 'withdrawal'){
+    return accumulator -= currentValue.amount
+  }
+  return accumulator += currentValue.amount
+}, 0);
+console.log(balance)
+
+const composite = traits.reduce((accumulator, currentValue) => {
+  for (let x in traits) {
```

Comment:
you should not be using a for in loop here. this should be done by the looping functionality of reduce.